### PR TITLE
feat: add report-missing-dependecy action

### PR DESCRIPTION
**Pull request by OndrejTuma**
[FX-3300]

### Description

Add new action for monorepo that report changed packages which are not in changeset.
The difference is reported as a PR comment.

### How to test

- you can check out the [Picasso PR](https://github.com/toptal/picasso/pull/3286#issuecomment-1350664508), push some changes to see how it behaves

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/davinci-github-actions/blob/master/_docs/contribution/changeset-guidelines.md) (if needed)

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

[FX-3300]: https://toptal-core.atlassian.net/browse/FX-3300?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: report-missing-changeset/action.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        BASE_BRANCH: ${{ inputs.base-branch }}
+      shell: bash
+      run: |
+        # Necessary for branch comparison: https://github.com/changesets/changesets/issues/517
+        git fetch origin $BASE_BRANCH
+
+    - name: Get changeset status
+      id: changeset-info
+      # TODO: Replace with official action once it's merged
+      uses: toptal/davinci-github-actions/get-changeset-info@feat/get-changeset-status
```
Comment (OndrejTuma) [inline report-missing-changeset/action.yml:31]: Reminder not to forget to replace with tag before merging

**Review by ozgurkececioglu — APPROVED** [review]
LGTM 👍 didn't test but checked the Picasso test PR

Thanks for this improvement on our dev process man 🙇‍♂️

**Review by TomasSlama — APPROVED** [review]
nice addition! 👍

**Review by mkrl — APPROVED** [review]
Looks great. Love the improvement

File: report-missing-changeset/action.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+              body,
+            })
+          } else {
+            await github.issues.createComment({
+              owner,
+              repo,
+              issue_number,
+              body,
+            })
+          }
```
Comment (dmaklygin) [inline report-missing-changeset/action.yml:95]: shouldn't we also put this code in a file?

File: report-missing-changeset/action.yml [hunk 3]
```diff
@@ -32,63 +32,10 @@ runs:
...
-              body,
-            })
-          }
+      shell: bash
+      env:
+        GITHUB_TOKEN: ${{ inputs.github-token }}
+        CHANGED_PACKAGES: ${{ steps.changeset-info.outputs.changed-packages }}
+        RELEASE_PACKAGES: ${{ steps.changeset-info.outputs.release-packages }}
+        MISSING_PACKAGES: ${{ steps.changeset-info.outputs.missing-packages }}
+      run: node ./report-missing-changeset/dist/index.js
```
Comment (dmaklygin) [inline report-missing-changeset/action.yml:41]: 👍

Comment (OndrejTuma) [reply report-missing-changeset/action.yml:41]: @dmaklygin I moved the logic to a separate file, please check it out 🙇

**Pull request merged**